strawberry/tools/depth_limit_validator.py

```diff
@@ -0,0 +1,171 @@
+import re
+from typing import Callable, List, Union
+
+from graphql import GraphQLError
+from graphql.language import (
+    DefinitionNode,
+    FieldNode,
+    FragmentDefinitionNode,
+    FragmentSpreadNode,
+    InlineFragmentNode,
+    OperationDefinitionNode,
+)
+from graphql.validation import ValidationContext, ValidationRule
+
+
+def depth_limit_validator(max_depth: int, options=None, callback=None):
```

Comment:
Should we add type hints for `options` and `callback`?

The type of `callback` is already part of the documentation. To properly type `options` we would probably have to use a TypedDict.

Reply:
Yes thats a good idea.